### PR TITLE
Fix duplicate check for nested dirs

### PR DESCRIPTION
**Pull request by MetRonnie**
PR against https://github.com/cylc/cylc-flow/pull/4442

### CODE REVIEW

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -1376,75 +1381,65 @@ def infer_latest_run(
...
+    if install_dir:
         # Search child tree for install directories:
-        search_patterns = [
-            f'*/{"*/"*depth}{WorkflowFiles.Install.DIRNAME}'
-            for depth in range(MAX_SCAN_DEPTH)
-        ]
-        for search_pattern in search_patterns:
-            results = list(path.glob(search_pattern))
-            for result in results:
-                LOG.critical(result)
```
Comment (MetRonnie) [inline cylc/flow/workflow_files.py:1440]: I take it this was lefotver debugging?

File: tests/functional/cylc-install/02-failures.t [hunk 2]
```diff
@@ -198,11 +198,10 @@ purge_rnd_workflow
 TEST_NAME="${TEST_NAME_BASE}-nested-rundir"
 make_rnd_workflow
 mkdir -p "${RND_WORKFLOW_RUNDIR}/.service"
-run_fail "${TEST_NAME}-install" cylc install --no-run-name \
-    -C "${RND_WORKFLOW_SOURCE}" \
```
Comment (MetRonnie) [inline tests/functional/cylc-install/02-failures.t:202]: I've reverted some of the test changes you made during the course of development as they aren't needed anymore

File: tests/unit/test_workflow_files.py [hunk 3]
```diff
@@ -119,17 +121,22 @@ def test_check_nested_dirs(tmp_run_dir: Callable):
...
         check_nested_dirs(test_dir)
-    assert "Nested run directories not allowed" in str(exc.value)
+    assert str(exc.value) == NESTED_DIRS_MSG.format(
+        dir_type='run', dest=test_dir, existing=(cylc_run_dir / 'a')
+    )
 
 
+@pytest.mark.parametrize(
+    'named_run', [True, False]
+)
```
Comment (MetRonnie) [inline tests/unit/test_workflow_files.py:131]: Important to test both named runs and `--no-run-name` installations

**Review by wxtim — APPROVED** [review]
Much cleaner than my code. Thanks.

**Pull request merged**